### PR TITLE
Add layout options to the appearance tab

### PR DESCRIPTION
**Pull request by JorikSchellekens**
Implemented:
Screen Recording 2020-06-16 at 18 03 07

Designs:
image

https://www.figma.com/file/HjeMwr2l0kfJDyP3IyiAeu/Text-Settings?node-id=0%3A1

### CODE REVIEW

**Comment by JorikSchellekens** [comment]
Most of the discrepancies here are due to font size differences

**Comment by t3chguy** [comment]
Given that it is behind labs it doesn't *need* Design review

File: src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx [hunk 1]
```diff
@@ -52,8 +52,11 @@ interface IState extends IThemeState {
     customThemeUrl: string,
     customThemeMessage: CustomThemeMessage,
     useCustomFontSize: boolean,
+    useIRCLayout: boolean,
 }
 
+const MESSAGE_PREVIEW_TEXT = "Hey you. You're the best"
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:58]: i18n?

File: src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx [hunk 2]
```diff
@@ -52,8 +52,11 @@ interface IState extends IThemeState {
     customThemeUrl: string,
     customThemeMessage: CustomThemeMessage,
     useCustomFontSize: boolean,
+    useIRCLayout: boolean,
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:55]: semi-colons retroactively plz :(

File: src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx [hunk 3]
```diff
@@ -67,6 +70,7 @@ export default class AppearanceUserSettingsTab extends React.Component<IProps, I
             customThemeUrl: "",
             customThemeMessage: {isError: false, text: ""},
             useCustomFontSize: SettingsStore.getValue("useCustomFontSize"),
+            useIRCLayout: SettingsStore.getValue("useIRCLayout", null),
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:73]: no need to specify the null

File: src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx [hunk 4]
```diff
@@ -323,6 +338,49 @@ export default class AppearanceUserSettingsTab extends React.Component<IProps, I
...
+            <div className="mx_AppearanceUserSettingsTab_Layout_RadioButtons" >
+                <div className="mx_AppearanceUserSettingsTab_Layout_RadioButton">
+                    <MessagePreview
+                        className="mx_AppearanceUserSettingsTab_Layout_RadioButton_preview"
+                        message={MESSAGE_PREVIEW_TEXT}
+                        useIRCLayout={true}
+                    />
+                    <StyledRadioButton
+                        name={"layout"}
+                        value={true}
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:357]: they'll get cast to strings, might as well start with them as such for clarity

**Comment by JorikSchellekens** [comment]
Thanks for those. Design will still have to review it when we move it from labs so I don't see any benefit it removing that review now

File: src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx [hunk 5]
```diff
@@ -35,27 +35,27 @@ interface IProps {
...
-    useIRCLayout: boolean,
+    fontSize: string;
+    customThemeUrl: string;
+    customThemeMessage: CustomThemeMessage;
+    useCustomFontSize: boolean;
+    useIRCLayout: boolean;
 }
 
-const MESSAGE_PREVIEW_TEXT = "Hey you. You're the best"
+const MESSAGE_PREVIEW_TEXT = _t("Hey you. You're the best!");
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:61]: this won't work as it'll be called before the language json files are loaded

Comment (JorikSchellekens) [reply src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:61]: Can I do it as a class member? Or instead _td here and _t everywhere else?

Comment (t3chguy) [reply src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx:61]: Depends on when the class is constructed, alternately the latter works too :)

**Pull request merged**